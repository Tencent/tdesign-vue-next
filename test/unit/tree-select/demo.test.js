/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import baseVue from '@/examples/tree-select/demos/base.vue';
import lazyVue from '@/examples/tree-select/demos/lazy.vue';
import prefixVue from '@/examples/tree-select/demos/prefix.vue';

const mapper = {
  baseVue,
  lazyVue,
  prefixVue,
};

describe('TreeSelect', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`TreeSelect ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
