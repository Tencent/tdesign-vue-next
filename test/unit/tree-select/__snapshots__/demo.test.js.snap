// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeSelect TreeSelect baseVue demo works fine 1`] = `
<div
  class="tdesign-tree-select-base"
>
  <div
    class="t-select__wrap"
    modelvalue=""
  >
    <div
      class="t-popup__reference t-select__popup-reference"
    >
      <transition-stub>
        <div
          aria-hidden="true"
          class="t-popup"
          role="tooltip"
          style="display: none;"
        >
          <div
            class="t-popup__content t-select__dropdown narrow-scrollbar"
          >
            
            <div>
              <p
                class="t-select-loading-tips"
                style="display: none;"
              >
                <div
                  class="t-select__empty"
                >
                  加载中
                </div>
              </p>
              <div
                class="t-tree t-tree--hoverable t-tree--transition t-tree--block-node"
                modelvalue=""
                size="medium"
              >
                <transition-group-stub>
                  
                  <div
                    class="t-tree__item t-tree__item--open"
                    data-level="0"
                    data-value="guangdong"
                    style="--level: 0;"
                  >
                    
                    <span
                      class="t-tree__icon t-folder-icon t-tree__icon--default"
                      ignore="active"
                      trigger="expand"
                    >
                      <svg
                        class="t-icon t-icon-caret-right-small"
                        fill="none"
                        height="1em"
                        viewBox="0 0 16 16"
                        width="1em"
                      >
                        <path
                          d="M6 5v6l4.5-3L6 5z"
                          fill="currentColor"
                          fill-opacity="0.9"
                        />
                      </svg>
                    </span>
                    <span
                      class="t-tree__label t-tree__label--strictly"
                    >
                      <span
                        style="position: relative;"
                      >
                        广东省
                      </span>
                    </span>
                    
                  </div>
                  <div
                    class="t-tree__item t-tree__item--open"
                    data-level="1"
                    data-value="guangzhou"
                    style="--level: 1;"
                  >
                    
                    <span
                      class="t-tree__icon t-folder-icon"
                      ignore="active"
                      trigger="expand"
                    >
                      
                    </span>
                    <span
                      class="t-tree__label t-tree__label--strictly"
                    >
                      <span
                        style="position: relative;"
                      >
                        广州市
                      </span>
                    </span>
                    
                  </div>
                  <div
                    class="t-tree__item t-tree__item--open"
                    data-level="1"
                    data-value="shenzhen"
                    style="--level: 1;"
                  >
                    
                    <span
                      class="t-tree__icon t-folder-icon"
                      ignore="active"
                      trigger="expand"
                    >
                      
                    </span>
                    <span
                      class="t-tree__label t-tree__label--strictly"
                    >
                      <span
                        style="position: relative;"
                      >
                        深圳市
                      </span>
                    </span>
                    
                  </div>
                  <div
                    class="t-tree__item t-tree__item--open"
                    data-level="0"
                    data-value="jiangsu"
                    style="--level: 0;"
                  >
                    
                    <span
                      class="t-tree__icon t-folder-icon t-tree__icon--default"
                      ignore="active"
                      trigger="expand"
                    >
                      <svg
                        class="t-icon t-icon-caret-right-small"
                        fill="none"
                        height="1em"
                        viewBox="0 0 16 16"
                        width="1em"
                      >
                        <path
                          d="M6 5v6l4.5-3L6 5z"
                          fill="currentColor"
                          fill-opacity="0.9"
                        />
                      </svg>
                    </span>
                    <span
                      class="t-tree__label t-tree__label--strictly"
                    >
                      <span
                        style="position: relative;"
                      >
                        江苏省
                      </span>
                    </span>
                    
                  </div>
                  <div
                    class="t-tree__item t-tree__item--open"
                    data-level="1"
                    data-value="nanjing"
                    style="--level: 1;"
                  >
                    
                    <span
                      class="t-tree__icon t-folder-icon"
                      ignore="active"
                      trigger="expand"
                    >
                      
                    </span>
                    <span
                      class="t-tree__label t-tree__label--strictly"
                    >
                      <span
                        style="position: relative;"
                      >
                        南京市
                      </span>
                    </span>
                    
                  </div>
                  <div
                    class="t-tree__item t-tree__item--open"
                    data-level="1"
                    data-value="suzhou"
                    style="--level: 1;"
                  >
                    
                    <span
                      class="t-tree__icon t-folder-icon"
                      ignore="active"
                      trigger="expand"
                    >
                      
                    </span>
                    <span
                      class="t-tree__label t-tree__label--strictly"
                    >
                      <span
                        style="position: relative;"
                      >
                        苏州市
                      </span>
                    </span>
                    
                  </div>
                  
                </transition-group-stub>
                <!---->
              </div>
            </div>
            
            <!---->
          </div>
        </div>
      </transition-stub>
      
      <div
        class="t-select t-select-polyfill t-size-m"
      >
        <!---->
        <span
          class="t-select__placeholder"
        >
          请选择
           
        </span>
        
        
        <span
          class="t-tag t-tag--default t-tag--dark t-size-m"
          style="display: none;"
        >
          <!---->
          
          +0
          
          <!---->
        </span>
        <!---->
        <div
          class="t-input__wrap t-select__input"
          modelvalue=""
          style="display: none;"
        >
          <div
            class="t-input"
          >
            <!---->
            <!---->
            <input
              class="t-input__inner"
              placeholder="请选择"
              type="text"
            />
            <!---->
            <!---->
            <!---->
          </div>
          <!---->
        </div>
        <svg
          class="t-fake-arrow t-select__right-icon t-select__right-icon-polyfill"
          fill="none"
          height="16"
          style="font-size: medium;"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
        <svg
          class="t-icon t-icon-close-circle-filled t-select__right-icon t-select__right-icon-polyfill t-select__active-icon"
          fill="none"
          height="1em"
          style="font-size: medium; display: none;"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M15 8A7 7 0 101 8a7 7 0 0014 0zM5.67 4.95L8 7.29l2.33-2.34.7.7L8.7 8l2.34 2.35-.71.7L8 8.71l-2.33 2.34-.7-.7L7.3 8 4.96 5.65l.71-.7z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
        <div
          class="t-loading--center t-size-s t-loading t-select__loading-tips t-select__right-icon-polyfill"
          style="display: none;"
        >
          <svg
            class="t-loading__gradient t-icon-loading"
            height="1em"
            size="small"
            version="1.1"
            viewBox="0 0 14 14"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <foreignobject
              height="12"
              width="12"
              x="1"
              y="1"
            >
              <div
                class="t-loading__gradient-conic"
              />
            </foreignobject>
          </svg>
          <!---->
        </div>
      </div>
      
    </div>
  </div>
</div>
`;

exports[`TreeSelect TreeSelect lazyVue demo works fine 1`] = `
<div
  class="tdesign-tree-select-lazy"
>
  <div
    class="t-select__wrap"
    modelvalue=""
  >
    <div
      class="t-popup__reference t-select__popup-reference"
    >
      <transition-stub>
        <div
          aria-hidden="true"
          class="t-popup"
          role="tooltip"
          style="display: none;"
        >
          <div
            class="t-popup__content t-select__dropdown narrow-scrollbar"
          >
            
            <div>
              <p
                class="t-select-loading-tips"
                style="display: none;"
              >
                <div
                  class="t-select__empty"
                >
                  加载中
                </div>
              </p>
              <div
                class="t-tree t-tree--hoverable t-tree--transition t-tree--block-node"
                modelvalue=""
                size="medium"
              >
                <transition-group-stub>
                  
                  <div
                    class="t-tree__item"
                    data-level="0"
                    data-value="1"
                    style="--level: 0;"
                  >
                    
                    <span
                      class="t-tree__icon t-folder-icon t-tree__icon--default"
                      ignore="active"
                      trigger="expand"
                    >
                      <svg
                        class="t-icon t-icon-caret-right-small"
                        fill="none"
                        height="1em"
                        viewBox="0 0 16 16"
                        width="1em"
                      >
                        <path
                          d="M6 5v6l4.5-3L6 5z"
                          fill="currentColor"
                          fill-opacity="0.9"
                        />
                      </svg>
                    </span>
                    <span
                      class="t-tree__label t-tree__label--strictly"
                    >
                      <span
                        style="position: relative;"
                      >
                        1
                      </span>
                    </span>
                    
                  </div>
                  <div
                    class="t-tree__item"
                    data-level="0"
                    data-value="2"
                    style="--level: 0;"
                  >
                    
                    <span
                      class="t-tree__icon t-folder-icon t-tree__icon--default"
                      ignore="active"
                      trigger="expand"
                    >
                      <svg
                        class="t-icon t-icon-caret-right-small"
                        fill="none"
                        height="1em"
                        viewBox="0 0 16 16"
                        width="1em"
                      >
                        <path
                          d="M6 5v6l4.5-3L6 5z"
                          fill="currentColor"
                          fill-opacity="0.9"
                        />
                      </svg>
                    </span>
                    <span
                      class="t-tree__label t-tree__label--strictly"
                    >
                      <span
                        style="position: relative;"
                      >
                        2
                      </span>
                    </span>
                    
                  </div>
                  
                </transition-group-stub>
                <!---->
              </div>
            </div>
            
            <!---->
          </div>
        </div>
      </transition-stub>
      
      <div
        class="t-select t-select-polyfill t-size-m"
      >
        <!---->
        <span
          class="t-select__placeholder"
        >
          请选择
           
        </span>
        
        
        <span
          class="t-tag t-tag--default t-tag--dark t-size-m"
          style="display: none;"
        >
          <!---->
          
          +0
          
          <!---->
        </span>
        <!---->
        <div
          class="t-input__wrap t-select__input"
          modelvalue=""
          style="display: none;"
        >
          <div
            class="t-input"
          >
            <!---->
            <!---->
            <input
              class="t-input__inner"
              placeholder="请选择"
              type="text"
            />
            <!---->
            <!---->
            <!---->
          </div>
          <!---->
        </div>
        <svg
          class="t-fake-arrow t-select__right-icon t-select__right-icon-polyfill"
          fill="none"
          height="16"
          style="font-size: medium;"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
        <svg
          class="t-icon t-icon-close-circle-filled t-select__right-icon t-select__right-icon-polyfill t-select__active-icon"
          fill="none"
          height="1em"
          style="font-size: medium; display: none;"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M15 8A7 7 0 101 8a7 7 0 0014 0zM5.67 4.95L8 7.29l2.33-2.34.7.7L8.7 8l2.34 2.35-.71.7L8 8.71l-2.33 2.34-.7-.7L7.3 8 4.96 5.65l.71-.7z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
        <div
          class="t-loading--center t-size-s t-loading t-select__loading-tips t-select__right-icon-polyfill"
          style="display: none;"
        >
          <svg
            class="t-loading__gradient t-icon-loading"
            height="1em"
            size="small"
            version="1.1"
            viewBox="0 0 14 14"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <foreignobject
              height="12"
              width="12"
              x="1"
              y="1"
            >
              <div
                class="t-loading__gradient-conic"
              />
            </foreignobject>
          </svg>
          <!---->
        </div>
      </div>
      
    </div>
  </div>
</div>
`;

exports[`TreeSelect TreeSelect prefixVue demo works fine 1`] = `
<div
  class="tdesign-tree-select-prefix"
>
  <div
    class="t-select__wrap"
    modelvalue=""
  >
    <div
      class="t-popup__reference t-select__popup-reference"
    >
      <transition-stub>
        <div
          aria-hidden="true"
          class="t-popup"
          role="tooltip"
          style="display: none;"
        >
          <div
            class="t-popup__content t-select__dropdown narrow-scrollbar"
          >
            
            <div>
              <p
                class="t-select-loading-tips"
                style="display: none;"
              >
                <div
                  class="t-select__empty"
                >
                  加载中
                </div>
              </p>
              <div
                class="t-tree t-tree--hoverable t-tree--transition t-tree--block-node"
                modelvalue=""
                size="medium"
              >
                <transition-group-stub>
                  
                  <div
                    class="t-tree__item t-tree__item--open"
                    data-level="0"
                    data-value="group1"
                    style="--level: 0;"
                  >
                    
                    <span
                      class="t-tree__icon t-folder-icon t-tree__icon--default"
                      ignore="active"
                      trigger="expand"
                    >
                      <svg
                        class="t-icon t-icon-caret-right-small"
                        fill="none"
                        height="1em"
                        viewBox="0 0 16 16"
                        width="1em"
                      >
                        <path
                          d="M6 5v6l4.5-3L6 5z"
                          fill="currentColor"
                          fill-opacity="0.9"
                        />
                      </svg>
                    </span>
                    <span
                      class="t-tree__label t-tree__label--strictly"
                    >
                      <span
                        style="position: relative;"
                      >
                        开发一组
                      </span>
                    </span>
                    
                  </div>
                  <div
                    class="t-tree__item t-tree__item--open"
                    data-level="1"
                    data-value="zhao"
                    style="--level: 1;"
                  >
                    
                    <span
                      class="t-tree__icon t-folder-icon"
                      ignore="active"
                      trigger="expand"
                    >
                      
                    </span>
                    <span
                      class="t-tree__label t-tree__label--strictly"
                    >
                      <span
                        style="position: relative;"
                      >
                        小赵
                      </span>
                    </span>
                    
                  </div>
                  <div
                    class="t-tree__item t-tree__item--open"
                    data-level="1"
                    data-value="qian"
                    style="--level: 1;"
                  >
                    
                    <span
                      class="t-tree__icon t-folder-icon"
                      ignore="active"
                      trigger="expand"
                    >
                      
                    </span>
                    <span
                      class="t-tree__label t-tree__label--strictly"
                    >
                      <span
                        style="position: relative;"
                      >
                        小钱
                      </span>
                    </span>
                    
                  </div>
                  <div
                    class="t-tree__item t-tree__item--open"
                    data-level="0"
                    data-value="group2"
                    style="--level: 0;"
                  >
                    
                    <span
                      class="t-tree__icon t-folder-icon t-tree__icon--default"
                      ignore="active"
                      trigger="expand"
                    >
                      <svg
                        class="t-icon t-icon-caret-right-small"
                        fill="none"
                        height="1em"
                        viewBox="0 0 16 16"
                        width="1em"
                      >
                        <path
                          d="M6 5v6l4.5-3L6 5z"
                          fill="currentColor"
                          fill-opacity="0.9"
                        />
                      </svg>
                    </span>
                    <span
                      class="t-tree__label t-tree__label--strictly"
                    >
                      <span
                        style="position: relative;"
                      >
                        开发二组
                      </span>
                    </span>
                    
                  </div>
                  <div
                    class="t-tree__item t-tree__item--open"
                    data-level="1"
                    data-value="sun"
                    style="--level: 1;"
                  >
                    
                    <span
                      class="t-tree__icon t-folder-icon"
                      ignore="active"
                      trigger="expand"
                    >
                      
                    </span>
                    <span
                      class="t-tree__label t-tree__label--strictly"
                    >
                      <span
                        style="position: relative;"
                      >
                        小孙
                      </span>
                    </span>
                    
                  </div>
                  <div
                    class="t-tree__item t-tree__item--open"
                    data-level="1"
                    data-value="li"
                    style="--level: 1;"
                  >
                    
                    <span
                      class="t-tree__icon t-folder-icon"
                      ignore="active"
                      trigger="expand"
                    >
                      
                    </span>
                    <span
                      class="t-tree__label t-tree__label--strictly"
                    >
                      <span
                        style="position: relative;"
                      >
                        小李
                      </span>
                    </span>
                    
                  </div>
                  
                </transition-group-stub>
                <!---->
              </div>
            </div>
            
            <!---->
          </div>
        </div>
      </transition-stub>
      
      <div
        class="t-select t-select-polyfill t-size-m t-has-prefix"
      >
        <span
          class="t-select__left-icon"
        >
          <svg
            class="t-icon t-icon-browse"
          >
            <use
              href="#t-icon-browse"
            />
          </svg>
        </span>
        <span
          class="t-select__placeholder"
        >
          请输入
           
        </span>
        
        
        <span
          class="t-tag t-tag--default t-tag--dark t-size-m"
          style="display: none;"
        >
          <!---->
          
          +0
          
          <!---->
        </span>
        <!---->
        <div
          class="t-input__wrap t-select__input"
          modelvalue=""
          style="display: none;"
        >
          <div
            class="t-input"
          >
            <!---->
            <!---->
            <input
              class="t-input__inner"
              placeholder="请输入"
              type="text"
            />
            <!---->
            <!---->
            <!---->
          </div>
          <!---->
        </div>
        <svg
          class="t-fake-arrow t-select__right-icon t-select__right-icon-polyfill"
          fill="none"
          height="16"
          style="font-size: medium;"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
        <svg
          class="t-icon t-icon-close-circle-filled t-select__right-icon t-select__right-icon-polyfill t-select__active-icon"
          fill="none"
          height="1em"
          style="font-size: medium; display: none;"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M15 8A7 7 0 101 8a7 7 0 0014 0zM5.67 4.95L8 7.29l2.33-2.34.7.7L8.7 8l2.34 2.35-.71.7L8 8.71l-2.33 2.34-.7-.7L7.3 8 4.96 5.65l.71-.7z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
        <div
          class="t-loading--center t-size-s t-loading t-select__loading-tips t-select__right-icon-polyfill"
          style="display: none;"
        >
          <svg
            class="t-loading__gradient t-icon-loading"
            height="1em"
            size="small"
            version="1.1"
            viewBox="0 0 14 14"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <foreignobject
              height="12"
              width="12"
              x="1"
              y="1"
            >
              <div
                class="t-loading__gradient-conic"
              />
            </foreignobject>
          </svg>
          <!---->
        </div>
      </div>
      
    </div>
  </div>
</div>
`;
